### PR TITLE
[4.0] Add router for com_config

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue #32950.

### Summary of Changes
This PR add routing support for com_config, solves the issue https://github.com/joomla/joomla-cms/issues/32950 . I must be honest that I still don't understand all the part component architecture in Joomla 4, so some of the code in this PR is just copy and modify. Hope I don't make any mistake.

### Testing Instructions
1. See https://github.com/joomla/joomla-cms/issues/32950 , confirm the issue
2. Apply patch, confirm the issue fixed.

### CODE REVIEW

File: administrator/components/com_config/src/Extension/ConfigComponent.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+\defined('JPATH_PLATFORM') or die;
+
+use Joomla\CMS\Component\Router\RouterServiceInterface;
+use Joomla\CMS\Component\Router\RouterServiceTrait;
+use Joomla\CMS\Extension\MVCComponent;
+
+/**
+ * Component class for com_config
+ *
+ * @since  __DEPLOY_VERSION
```
Comment (Quy) [inline administrator/components/com_config/src/Extension/ConfigComponent.php]:
```suggestion
 * @since  __DEPLOY_VERSION__
```

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on de7beaa72e8b7d088e69c6f16890678321a84e7b<br /><br />Thanks very much<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32960">issues.joomla.org/tracker/joomla-cms/32960</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on de7beaa72e8b7d088e69c6f16890678321a84e7b<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32960">issues.joomla.org/tracker/joomla-cms/32960</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/32960">issues.joomla.org/tracker/joomla-cms/32960</a>.</sub>

**Comment by wilsonge** [comment]
Thanks!

**Comment by sandewt** [comment]
Oops, was the next code review visible to you?


Why

```
function (Container $container) {

```
and not as usual?
```
function (Container $container)
{
```

**Comment by joomdonation** [comment]
Thanks @sandewt. I made https://github.com/joomla/joomla-cms/pull/33046 to fix this code style issue.

**Pull request merged**